### PR TITLE
Add contributing doc

### PR DESCRIPTION
**Pull request by SeanTAllen**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+* [Pull request](#pull-request)
+
+Additional notes regarding formatting:
+
+* [Documentation formatting](#documentation-formatting)
+* [Code formatting](#code-formatting)
+
+Feature request
+---------------
+For any feature requests or enhancements to Corral, it is quite likely that you have to go through our [RFC process](https://github.com/ponylang/rfcs). Before opening or submitting any feature requests, please make sure you are familiar with the RFC process and follow the process as required.
```
Comment (cquinn) [inline CONTRIBUTING.md]: Do we want to require all Corral features + enhancements to go through the Pony RFC process? How about just larger changes, where the threshold can be discussed on Zulip.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+----------
+First of all please [search existing issues](https://github.com/ponylang/corral/issues) to make sure your issue hasn't already been reported. If you cannot find a suitable issue — [create a new one](https://github.com/ponylang/corral/issues/new).
+
+Provide the following details:
+
+  - short summary of what you was trying to achieve,
+  - a code causing the bug,
+  - expected result,
+  - actual results and
+  - environment details: at least operating system and compiler version (`ponyc -v`).
```
Comment (cquinn) [inline CONTRIBUTING.md]: Maybe we should add a version to Corral that is easily available on the command line and request that here too.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+* Ensure that the ticket title is appropriate - the title will be used as the summary of the change, so it should be appropriately formatted, including a ticket reference if the PR is a fix to an existing bug ticket.
+  * For example, an appropriate title for a PR that fixes a bug reported in issue ticket #98 might look like:
+  * *Fixed compiler crash related to tuple recovery (issue #98)*
+
+Once those conditions are met, the PR can be merged, and an automated system will immediately add the entry to the changelog. Keeping the changelog entries out of the file changes in the PR helps to avoid conflicts and other administrative headaches when many PRs are in progress.
+
+Please note, if your changes are purely to things like README, CHANGELOG etc, you can add [skip ci] as the last line of your commit message and your PR won't be run through our continuous integration systems. We ask that you use [skip ci] where appropriate as it helps to get changes through CI faster and doesn't waste resources that Appveyor and TravisCI are kindly donating to the Open Source community.
```
Comment (cquinn) [inline CONTRIBUTING.md]: Have we tried using labels to indicate [skip ci] instead of message text?

Comment (SeanTAllen) [reply CONTRIBUTING.md]: That's not how the ci systems work according to their docs.

File: CONTRIBUTING.md (hunk 2 above)
Comment (SeanTAllen) [reply CONTRIBUTING.md]: Agreed. Can you open a PR for that?

File: CONTRIBUTING.md (hunk 1 above)
Comment (SeanTAllen) [reply CONTRIBUTING.md]: That is what the language generally says, it doesn't say required. Say "likely". This is the standard boilerplate for all the Pony stuff.

File: CONTRIBUTING.md (hunk 2 above)
Comment (cquinn) [reply CONTRIBUTING.md]: OK, I will work on that.

**Pull request merged**